File: src/main/java/szewek/mcflux/util/MCFluxLocation.java
package szewek.mcflux.util;

import net.minecraft.util.ResourceLocation;

import javax.annotation.Nonnull;

import static szewek.mcflux.R.MF_NAME;

public final class MCFluxLocation extends ResourceLocation {
	private static final int MF_HASH = MF_NAME.hashCode() * 31;

	public MCFluxLocation(String name) {
		super(0, MF_NAME, name);
	}

	@Nonnull @Override
	public String getResourceDomain() {
		return MF_NAME;
	}

	@Nonnull @Override
	public String toString() {
		return MF_NAME + ':' + resourcePath;
	}

	@Override
	public int hashCode() {
		return MF_HASH + resourcePath.hashCode();
	}
}

File: src/main/java/szewek/mcflux/util/MCFluxLocation.kt
package szewek.mcflux.util

import net.minecraft.util.ResourceLocation

import szewek.mcflux.R.MF_NAME

class MCFluxLocation(name: String) : ResourceLocation(0, MF_NAME, name) {

	override fun getResourceDomain(): String {
		return MF_NAME
	}

	override fun toString(): String {
		return MF_NAME + ':'.toString() + resourcePath
	}

	override fun hashCode(): Int {
		return MF_HASH + resourcePath.hashCode()
	}

	override fun equals(other: Any?): Boolean {
		if (other !is ResourceLocation)
			return false
		return other.resourceDomain == MF_NAME && other.resourcePath == resourcePath
	}

	companion object {
		private val MF_HASH = MF_NAME.hashCode() * 31
	}
}
